leakcanary-android-core/src/main/java/leakcanary/internal/activity/screen/HeapDumpRenderer.kt

```diff
@@ -101,181 +101,178 @@ internal object HeapDumpRenderer {
 
     var lastPosition = 0L
 
-    Hprof.open(heapDumpFile)
-        .use { hprof ->
-          val hprofStringCache = mutableMapOf<Long, String>()
-          val classNames = mutableMapOf<Long, Long>()
-          hprof.reader.readHprofRecords(
-              setOf(HprofRecord::class), OnHprofRecordListener { position, record ->
-            lastPosition = position
-            when (record) {
-              is StringRecord -> {
-                hprofStringCache[record.id] = record.string
-              }
-              is LoadClassRecord -> {
-                classNames[record.id] = record.classNameStringId
-              }
-            }
-            val localCurrentRecord = currentRecord
-            when {
-              localCurrentRecord is HeapDumpInfoRecord -> {
-                val colorForHeapInfo =
-                  when (hprofStringCache[localCurrentRecord.heapNameStringId]) {
-                    // The primary heap on which your app allocates memory.
-                    "app" -> appHeapColor
-                    // The system boot image, containing classes that are preloaded during boot time.
-                    // Allocations here are guaranteed to never move or go away.
-                    "image" -> imageHeapColor
-                    // The copy-on-write heap where an app process is forked from in the Android system.
-                    "zygote" -> zygoteHeapColor
-                    // JNI heap: The heap that shows where Java Native Interface (JNI) references are allocated and released.
-                    // default heap: When no heap is specified by the system
-                    else -> otherColor
-                  }
-                recordPositions.add(colorForHeapInfo to position)
-                currentRecord = record
-              }
-              localCurrentRecord is InstanceDumpRecord
-                  && hprofStringCache[classNames[localCurrentRecord.classId]] == "java.lang.String"
-                  && (record !is InstanceDumpRecord || hprofStringCache[classNames[record.classId]]
-                  != "java.lang.String")
-              -> {
-                recordPositions.add(stringColor to position)
-                currentRecord = record
-              }
-              currentRecord == null -> {
-                recordPositions.add(otherColor to position)
-                currentRecord = record
-              }
-              currentRecord!!::class != record::class -> {
-                recordPositions.add(colors.getValue(currentRecord!!::class) to position)
-                currentRecord = record
-              }
+    val reader = StreamingHprofReader.readerFor(heapDumpFile)
```

Comment:
![image](https://user-images.githubusercontent.com/2152942/92329027-5eeb6800-f02a-11ea-9197-bd31d2b77d10.png)

Ok, hiding whitespaces made it easier to review!